bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java

```diff
@@ -266,9 +266,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                             break;
                         }
                         case CHARGE_AMPS:
-                            if (command instanceof QuantityType) {
-                                @SuppressWarnings({ "unchecked", "null" })
-                                int amps = ((QuantityType<ElectricCurrent>) command).toUnit(Units.AMPERE).intValue();
+                            if (command instanceof DecimalType) {
+                                int amps = ((DecimalType) command).intValue();
+                                if (command instanceof QuantityType<?>) {
```

Comment:
Sorry but I believe it is not good. There is no hierarchy between DecimalType and QuanyityType. 
You have to separate the two cases.

Reply:
Thanks for looking so closely! You're clearly right, I have fixed this. Hope this time I got it correct!